core/lib/zksync_core/src/consensus/mod.rs

```diff
@@ -0,0 +1,124 @@
+use anyhow::Context as _;
+use zksync_concurrency::{ctx, time};
+use zksync_consensus_roles::validator;
+use zksync_types::block::ConsensusBlockFields;
+use zksync_types::{Address, MiniblockNumber};
+
+mod payload;
+mod proto;
+
+pub(crate) use payload::Payload;
+
+struct PostgresState(zksync_dal::ConnectionPool);
+
+#[async_trait::async_trait]
+pub trait BlockChainState {
```

Comment:
Is it possible to remove this trait from this PR? It seems unrelated to adding the `consensus` field for Postgres, doesn't look necessary, and it raises some questions. The main question is about the data flow: can we afford non-atomic update of consensus block fields for an already-persisted L2 block? Right now, my PR (#371) isn't accounting for this possibility (note that saving consensus fields is built into miniblock persistence logic and is atomic with saving the miniblock header itself), and in general it raises some concerns: What if a node crashes after persisting a block, but before adding consensus fields for it? What if another node requests a block in such a state? It's not that these issues are insurmountable, but IMO, atomicity makes things quite a bit easier.

Reply:
yeah, it's unrelated. Removing.